Java Problems Solved

obedjamir opened the pull request:
Java problems solved.

obedjamir committed:
Java Problems Solved

PinakiGhosh commented on Question1.java:
```diff
@@ -0,0 +1,65 @@
+    public String Age;
+    public String Phone_number;
+    public String Address;
+    public String Salary;
```
Java variable naming convention is
- Starts with a lower case
- Use capitalisation to denote different words
Ex dataMembers, age etc

obedjamir replied:
This issue has been resolved.

PinakiGhosh commented on Question2.java:
```diff
@@ -0,0 +1,137 @@
+        Node next;
+    }
+
+    public void add_end(Type data) //Add to end of list
```
In Java we dont use _ different words are separated by capitalisations.
Ex addEnd

obedjamir replied:
This issue has been resolved.

PinakiGhosh commented on Question2.java:
```diff
@@ -0,0 +1,137 @@
+        Node next;
+    }
+
+    public void add_end(Type data) //Add to end of list
```
For comments use the following format
/**
   * Add to the end of the list.
   * 
   * @param data - The data to be inserted
   */

obedjamir replied:
This issue has been resolved.

PinakiGhosh commented on Question2.java:
```diff
@@ -0,0 +1,137 @@
+        node.data = data;
+        node.next = null;
+
+        if(head==null) head = node;
```
Give braces even for single line if.

obedjamir replied:
This issue has been resolved.

PinakiGhosh commented on Question2.java:
```diff
@@ -0,0 +1,137 @@
+{
+    Node head;
+
+    private class Node //Create node
```
Create node as a separate class with getter and setters for accessing data and next

PinakiGhosh replied:
Also add a constructor which takes the data as input and creates the node with the given data

obedjamir replied:
This issue has been resolved.

PinakiGhosh commented on Question2.java:
```diff
@@ -0,0 +1,137 @@
+
+        else
+        {
+            Node cursor = head;
```
This initialisation will actually modify head whenever cursor is modified. Try it out in local once.

PinakiGhosh commented on Question2.java:
```diff
@@ -0,0 +1,137 @@
+        head = head.next;
+    }
+
+    public void delete_end() //Delete from end of list
```
This will need 2 iterations for deleting the node. Can we do it in a single iteration?

PinakiGhosh commented on Question2.java:
```diff
@@ -0,0 +1,137 @@
+            }
+            System.out.println(node.data);
+        }
+        catch(NullPointerException e)
```
Instead of try catch do a check first and print **List is empty!** if there are no elements. Else print the list.

PinakiGhosh commented on Question3.java:
```diff
@@ -0,0 +1,125 @@
+}
+
+//card_select class
+class card_select
```
Class name should always start with capital letters. No underscore in between

PinakiGhosh commented on Question3.java:
```diff
@@ -0,0 +1,125 @@
+}
+
+//card_select class
+class card_select
```
Have 1 class named Card.
It will have data members

1. suite
2. number

Have a constructor which takes both these values and does the validation.

Make the card class implement Comparable interface and have the sorting logic there.

In your main method create a list of cards of same suite and see if the sorting works with Collection.sort method
Try and compare 2 cards of different suite and see what happens.

Dont need to take input from user. Make static objects

obedjamir replied:
This issue has been resolved.

PinakiGhosh commented on Question5.java:
```diff
@@ -0,0 +1,152 @@
+    {
+        if(max <= 0) //Validate Argument
+        {
+            throw new IllegalArgumentException("Illegal Argument");
```
Change the message to Size cannot be {given size}

obedjamir replied:
This issue has been resolved.

PinakiGhosh commented on Question5.java:
```diff
@@ -0,0 +1,152 @@
+//Limited Stack Class
+class LimitedStack<Type> implements Stack<Type>
+{
+    Node head;
```
Data members should be private

obedjamir replied:
This issue has been resolved.

PinakiGhosh commented on Question5.java:
```diff
@@ -0,0 +1,152 @@
+    }
+
+    @Override
+    public int size() //Size method
```
Can we keep a counter and update during push/pop to track the size instead of iterating the entire loop everytime?

obedjamir replied:
This issue has been resolved.

PinakiGhosh commented on Question5.java:
```diff
@@ -0,0 +1,152 @@
+    }
+    catch(NullPointerException e)
+    {
+        System.out.println("Stack is empty");
```
Dont catch exception. Add a check and print accordingly.

obedjamir replied:
This issue has been resolved.

obedjamir committed:
Java problems updates
- Data Members removed from Question1
- Naming convention updated for all files
- Parenthesis added for if/else with single statement

obedjamir committed:
Java problems updates
- Data Members removed from Question1
- Naming convention updated for all files
- Parenthesis added for if/else with single statement

obedjamir committed:
Java problems updates
- Data Members removed from Question1
- Naming convention updated for all files
- Parenthesis added for if/else with single statement
- Reported issues resolved

obedjamir committed:
Java problems updates

- Data Members removed from Question1
- Naming convention updated for all files
- Parenthesis added for if/else with single statement
- Reported issues resolved
- Resolved size() counter in Question5
